### ISSUE TITLE
GetReminder throws exception when reminder don't exists

### ISSUE THREAD

**Issue by EranOfer** [dotnet/orleans#1167]
Hi all,
I believe there is a bug in `LocalReminderService.GetReminder`.
When I call `Grain.GetReminder` with a reminder that doesn't exist, I get an exception (Null reference exception).
This is a common scenario if you need to unregister a reminder and you don't know if it exists because of a reactivation (That made you lose the local reference to IGrainReminder). 

Please look at `LocalReminderService.GetReminder`:

``` C#
    public async Task<IGrainReminder> GetReminder(GrainReference grainRef, string reminderName)
        {
            if(logger.IsVerbose) logger.Verbose(ErrorCode.RS_GetReminder,"GetReminder: GrainReference={0} ReminderName={1}", grainRef.ToDetailedString(), reminderName);
            var entry = await reminderTable.ReadRow(grainRef, reminderName);
            return entry.ToIGrainReminder();
        }
```
`entry` can be **null**

```
 internal async Task<ReminderEntry> ReadReminderRowAsync(string serviceId, GrainReference    grainRef, string reminderName)
{
.....
    return ret != null ? Enumerable.FirstOrDefault<ReminderEntry>(ret) : (ReminderEntry) null;
}
```
This function can return **null**

I would propose to check if the entry is null before accessing its properties, and if it is so then just  return null.

**Comment by gabikliot** [comment]
Yes, I think you are correct.
Would you like to submit a PR with the fix?

**Comment by EranOfer** [comment]
open PR: https://github.com/dotnet/orleans/pull/1182

**Comment by gabikliot** [comment]
Fixed via #1182 .
